crates/subspace-networking/src/request_handlers/generic_request_handler.rs

```diff
@@ -53,23 +51,19 @@ pub type ExternalRequestHandler<Request> = Arc<
 pub(crate) struct GenericRequestHandler<Request: GenericRequest> {
     request_receiver: mpsc::Receiver<IncomingRequest>,
     request_handler: ExternalRequestHandler<Request>,
-    log_target: &'static str,
-    protocol_name: &'static str,
     protocol_config: ProtocolConfig,
 }
 
 impl<Request: GenericRequest> GenericRequestHandler<Request> {
-    pub fn new(handler_config: RequestHandlerConfig<Request>) -> Self {
+    pub fn new(handler_config: GenericRequestHandlerConfig<Request>) -> Self {
         let (request_sender, request_receiver) = mpsc::channel(REQUESTS_BUFFER_SIZE);
 
-        let mut protocol_config = ProtocolConfig::new(handler_config.protocol_name.into());
+        let mut protocol_config = ProtocolConfig::new(Cow::Borrowed(Request::PROTOCOL_NAME));
```

Comment:
Maybe we should also remove `Cow` and just use `&'static str`. Wdut?

Reply:
It depends on whether we want to create protocols on the fly or not. I guess we don't want that, will address now.